desci-contracts/test/DpidAliasRegistry.ts

```diff
@@ -0,0 +1,227 @@
+import { expect, use } from "chai";
+import chaiAsPromised from "chai-as-promised";
+import { ethers as hhe, upgrades } from "hardhat";
+import { BigNumber, ContractReceipt, ContractTransaction } from "ethers";
+import {
+  DpidAliasRegistry__factory,
+  DpidAliasRegistry,
+} from "../typechain-types";
+import { TransactionReceipt } from "@ethersproject/abstract-provider";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+use(chaiAsPromised);
+
+const FIRST_DPID = hhe.BigNumber.from(100);
+
+describe("dPID", () => {
+  let _accounts: SignerWithAddress[];
+  let owner: SignerWithAddress;
+  let user1: SignerWithAddress;
+  let user2: SignerWithAddress;
+  let DpidAliasRegistryFactory: DpidAliasRegistry__factory;
+  let dpidAliasRegistry: DpidAliasRegistry;
+
+  before( async () => {
+    _accounts = await hhe.getSigners()
+    owner = _accounts[0];
+    user1 = _accounts[1];
+    user2 = _accounts[2];
+
+    DpidAliasRegistryFactory = await hhe.getContractFactory(
+      "DpidAliasRegistry",
+    ) as DpidAliasRegistry__factory;
+
+    dpidAliasRegistry = await upgrades.deployProxy(
+      DpidAliasRegistryFactory,
+      [
+        FIRST_DPID // firstDpid
+      ],
+      {
+        initializer: "__DpidAliasRegistry_init"
+      }
+    ) as DpidAliasRegistry;
+    await dpidAliasRegistry.deployed();
+
+    // Default instance to non-owner user
+    dpidAliasRegistry = dpidAliasRegistry.connect(user1);
+  });
+
+  describe("deployment", () => {
+    let reciept: TransactionReceipt;
+    let proxyAddress: string;
+    let implAddress: string;
+    let proxyAdmin: string;
+
+    before(async () => {
+      reciept = await dpidAliasRegistry.deployTransaction.wait();
+      proxyAddress = reciept.contractAddress;
+      implAddress = await upgrades.erc1967.getImplementationAddress(proxyAddress);
+      proxyAdmin = await upgrades.erc1967.getAdminAddress(proxyAddress);
+
+      console.log({
+        implAddress,
+        proxyAdmin,
+        implOwner: await dpidAliasRegistry.owner(),
+      })
+    });
+
+    it("costs a reasonable amount of gas", async () => {
+      expect(
+        BigNumber.from(reciept.cumulativeGasUsed).lte(32000000),
+        `Gas limit exceeded`
+      ).to.be.true;
+    });
+
+    it("is made through proxy", () => {
+      expect(proxyAddress).not.to.equal(implAddress);
+    });
+
+    it("deploys implementation with proxy admin as owner", async () => {
+      const registryOwner = await dpidAliasRegistry.owner();
+      expect(registryOwner).to.equal(proxyAdmin);
+    });
+
+    it("respects initializer dpid offset", async () => {
+      const nextDpid = await dpidAliasRegistry.nextDpid();
+      expect(nextDpid).to.equal(FIRST_DPID);
+    })
+  });
+
+  describe("alias registry", () => {
+    const STREAM_A = "kjzl6kcym7w8y7i5ugaq9a3vlm7hhuaf4bpl5o5qykeh4qtsa12c6rb5ekw6aaa";
+    
+    describe("entry", () => {
```

Comment:
All positive tests to check code correctness. I'd advise you write tests for exceptional cases like
`it("should not mint dpid for a legacy dpid")`
`it("should not mint dpid for an umigrated legacy dpid")`